### PR TITLE
Create .codeclimate.yml

### PR DESCRIPTION
**Pull request by davehenton**

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- "features/"
+- "**/node_modules/"
+- "script/"
+- "**/spec/"
+- "**/test/"
+- "**/tests/"
+- "Tests/"
+- "**/vendor/"
+- "**/*_test.go"
+- "**/*.d.ts"
```
Comment (mrchief) [inline .codeclimate.yml]: I think I need these exclude patterns:

```
exclude_patterns:
  - "docs/"
  - "snapshots/"
  - "**/node_modules/"
  - "**/*.test.js"
```

**Comment by davehenton** [comment]
Hey @mrchief, quick update here. After looking at the docs for the eslint-plugin-react plugin [here](https://github.com/yannickcr/eslint-plugin-react#configuration), it seems that you need to explicitly enable JSX support in your ESLint config in order to define the rules properly. I've updated the `eslintrc.json` to account for that, and I've also updated the exclude patterns to reflect what you mentioned. Let me know if this works for you!

**Comment by mrchief** [comment]
@davehenton Thanks for this update. airbnb config is supposed to enable JSX support, hence the omission. But having it here doesn't harm so 🤷‍♂️ 

Also, I had to disable the object curly rule to get around CC errors. When you merged master, you got this rule merged as well: `"object-curly-newline": 0`. 

Do you mind reverting this rule and pushing again (to ensure that the issue has indeed been fixed)?

**Pull request merged**